lib/block-patterns.php

```diff
@@ -186,9 +186,8 @@
 // Deactivate the legacy patterns bundled with WordPress, and add new block patterns for testing.
 // More details in the trac issue (https://core.trac.wordpress.org/ticket/52846).
 add_action(
-	'init',
+	'admin_init',
```

Comment:
I don't think this should change, but if I leave it as `init`, I get this error when running the unit tests:

> Fatal error: Uncaught Error: Class 'Spy_REST_Server' not found in /var/www/html/wp-includes/rest-api.php:509

This was happening locally and in [the PR's unit test action](https://github.com/WordPress/gutenberg/runs/2403775789).

Reply:
Would it be worth using a more restrictive hook so that the API call isn't being made on admin screens that don't use patterns? In WordPress.com, where we use [a similar approach](https://github.com/Automattic/wp-calypso/blob/94c1213c164e6e2e6f9d5ecd19ff769a5688303c/apps/editing-toolkit/editing-toolkit-plugin/full-site-editing-plugin.php#L247) to load patterns from an API endpoint we decided to use the `current_screen` hook and check for `is_block_editor` or `gutenberg_is_edit_site_page()` so that we only call the endpoint on views that need it.

Reply:
@andrewserong has a good point on employing a more restrictive hook, Before we used an API call it didn't matter, but now it does. Also, we're prefetching a lot of data in the Gutenberg editors. Shouldn't this be treated the same way?

Reply:
This data isn't used the same way as the other prefetched data - it's not used in gutenberg directly, it's only used on the server-side to register patterns. Once the pattern directory is more dynamic, and block patterns are registered in JS, we can switch to something more like the other prefetched endpoints.

In 89045a6b3755c80f978f8c7329b0b18b6267f921, I updated the code here to use the `current_screen` approach for loading pattern content, but left the unregistration on `init`, to match what's happening in core now.